src/Comsave/SafeSalesforceSaverBundle/Consumers/SalesforceSaverConsumer.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Comsave\SafeSalesforceSaverBundle\Consumers;
+
+use Comsave\SafeSalesforceSaverBundle\Factory\ExceptionMessageFactory;
+use Comsave\SafeSalesforceSaverBundle\Services\ModelSerializer;
+use LogicItLab\Salesforce\MapperBundle\MappedBulkSaver;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerInterface;
+
+class SalesforceSaverConsumer implements ConsumerInterface
+{
+    /** @var MappedBulkSaver */
+    private $mappedBulkSaver;
+
+    /** @var ModelSerializer */
+    private $modelSerializer;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    /**
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        MappedBulkSaver $mappedBulkSaver,
+        ModelSerializer $modelSerializer,
+        LoggerInterface $logger
+    ) {
+        $this->mappedBulkSaver = $mappedBulkSaver;
+        $this->modelSerializer = $modelSerializer;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param AMQPMessage $message
+     * @return array|null
+     * @throws \Throwable
+     */
+    public function execute(AMQPMessage $message)
+    {
+        $this->logger->debug(ExceptionMessageFactory::build($this, implode('. ', [
+            'Consuming',
+            $message->body
+        ])));
+
+        $models = $this->unserializeModels($message);
+
+        try {
+            foreach ($models as $model) {
+                $this->mappedBulkSaver->save($model);
+            }
+
+            $result = $this->mappedBulkSaver->flush();
+        } catch (\Throwable $ex) {
+            $this->logger->error(ExceptionMessageFactory::build($this, implode('. ', [
```

Comment:
It would be nice to add the exception message detection to this as well. 

When we can't save an object because of some missing field then retrying is fairly useless. So we can log the exception and then not rethrow it. But if the exception message contains either `org is locked` or `unable to obtain exclusive access` then it is a good idea to throw the exception so that it gets requeued.